python/ray/data/_internal/execution/interfaces/common.py

```diff
@@ -1,2 +1,181 @@
+import bisect
+import json
+from typing import Dict, List, Tuple
+
+from ray.util.metrics import Histogram
+
 # Node id string returned by `ray.get_runtime_context().get_node_id()`.
 NodeIdStr = str
+
+# Used for time-based histograms (e.g. task completion time, block completion time)
+histogram_buckets_s = [
+    0.1,
+    0.25,
+    0.5,
+    1.0,
+    2.5,
+    5.0,
+    7.5,
+    10.0,
+    15.0,
+    20.0,
+    25.0,
+    50.0,
+    75.0,
+    100.0,
+    150.0,
+    500.0,
+    1000.0,
+    2500.0,
+    5000.0,
+]
+
+KiB = 1024
+MiB = 1024 * KiB
+GiB = 1024 * MiB
+
+# Used for size-based histograms (e.g. block size in bytes)
+histogram_buckets_bytes = [
+    KiB,
+    8 * KiB,
+    64 * KiB,
+    128 * KiB,
+    256 * KiB,
+    512 * KiB,
+    MiB,
+    8 * MiB,
+    64 * MiB,
+    128 * MiB,
+    256 * MiB,
+    512 * MiB,
+    GiB,
+    4 * GiB,
+    16 * GiB,
+    64 * GiB,
+    128 * GiB,
+    256 * GiB,
+    512 * GiB,
+    1024 * GiB,
+    4096 * GiB,
+]
+
+# Used for row count-based histograms (e.g. block size in rows)
+histogram_bucket_rows = [
+    1,
+    5,
+    10,
+    25,
+    50,
+    100,
+    250,
+    500,
+    1000,
+    2500,
+    5000,
+    10000,
+    25000,
+    50000,
+    100000,
+    250000,
+    500000,
+    1000000,
+    2500000,
+    5000000,
+    10000000,
```

Comment:
JFYI: you can write such literals like hits for readability 10_000_000 (no need to fix)